realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -102,7 +102,8 @@
     public void realmSet$columnString(String value) {
         ((RealmObject) this).realm.checkIfValid();
         if (value == null) {
-            throw new IllegalArgumentException("Trying to set non-nullable field columnString to null.");
+            ((RealmObject) this).row.setNull(columnInfo.columnStringIndex);
+            return;
```

Comment:
I also heavily lean toward `else` instead `return` but this is generated with [`ClassMetaData.isNullable()`](https://github.com/realm/realm-java/blob/master/realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java#L240) that is slightly out of the PR's scope. 

I'll look into refactor generator in another PR.